std/machines/hash/poseidon_bb.asm

```diff
@@ -0,0 +1,206 @@
+use std::array;
+use std::check::assert;
+use std::utils::unchanged_until;
+use std::utils::force_bool;
+use std::utils::sum;
+use std::convert::expr;
+use std::machines::memory16::Memory16;
+use std::machines::split::split_bb::SplitBB;
+
+// Implements the Poseidon permutation for the BabyBear field with the following parameters:
+// - 16 field elements in the state, of which
+//   - 8 are meant to be rate elements
+//   - 8 are meant to be capacity elements
+// - 8 field elements of output
+// - 8 full rounds
+// - 16 partial rounds
+// - S-Box: x^7
+//
+// The the number of rounds to get 128-bit security was taken from here:
+// https://github.com/Plonky3/Plonky3/blob/2df15fd05e2181b31b39525361aef0213fc76144/poseidon2/src/round_numbers.rs#L42
+//
+// This version of the Poseidon machine receives memory pointers and interacts
+// with memory directly to fetch its inputs and write its outputs.
+machine PoseidonBB(mem: Memory16, split_bb: SplitBB) with
+    latch: CLK_0,
+    operation_id: operation_id,
+    // Allow this machine to be connected via a permutation
+    call_selectors: sel,
+{
+
+    // Hashes 16 "rate" elements and 8 "capacity" elements to 8 field elements
```

Comment:
the comments in this file need to be revised... i think there are leftovers from the GL version

Reply:
I am not sure what is the best way to fix this, because that is the terminology in the original Poseidon paper, that assumed the permutation would be used in sponge construction. This new way of using the permutation directly on a Merkle tree is from Poseidon2 paper.

Reply:
Ah, the numbers don't even add up.

I fixed by removing confusing text.